lib/elixir/lib/string.ex

```diff
@@ -488,6 +488,12 @@ defmodule String do
       iex> String.split(String.normalize("é", :nfc), "e")
       ["é"]
 
+  When using both the `:trim` and the `:parts` option, note that the
+  trimming doesn't affect the last part:
+
+      iex> String.split(" a  b  c  ", " ", trim: true, parts: 2)
+      ["a", " b  c  "]
+
```

Comment:
I wouldn't say that trimming doesn't affect the last part. Rather, we return two parts and, because of that, there are no empty strings in the result. It is easy if you see the result of `parts: 2` first:

```
String.split("1  2  3", " ", [parts: 2])
["1", " 2  3"]
```

```suggestion
  When using both the `:trim` and the `:parts` option, first all
  parts are computed and then they are trimmed (if any):

      iex> String.split(" a  b  c  ", " ", trim: true, parts: 2)
      ["a", " b  c  "]
      iex> String.split(" a  b  c  ", " ", trim: true, parts: 3)
      ["a", "b  c  "]

```

Reply:
I understand but this explanation might still feel confusing due to the different meaning of `trim` here (removing empty strings) and in `String.trim`.

Perhaps something like
> When using both the `:trim` and the `:parts` option, first all
> parts are computed and then the empty ones are removed (if any):

_(side-discussion: I occasionally forget what trim means in this context and get confused, wondered if we could rename it but I don't have any good naming idea... `no_empty: true`?)_

Reply:
Good point, I have updated my suggestion above.

Reply:
I don't think I can update a suggestion, but the last value is wrong I think?

```elixir
String.split(" a  b  c  ", " ", trim: true, parts: 3)
["a", "b", " c  "]
```

Reply:
Well, if the last value is wrong then my explanation is wrong 🤦  My bad. Let me adjust it all.

Reply:
Ok, it should be good now. In any case, feel free to accept my suggestion and then modify it :)

Reply:
Very clear ✨ :100: